src/chapters/unit-testing/unit-testing-in-action.rst

```diff
@@ -0,0 +1,218 @@
+
+   describe("isPalindrome", function(){
+
+      // TODO - write some tests!
+
+   });
+
+Okay, let's write some tests!
+
+Positive and Negative Test Cases
+--------------------------------
+
+Positve Test Cases
+^^^^^^^^^^^^^^^^^^
+
+We'll start with positive and negative tests. For ``isPalindrome``, some positive tests have inputs:
+
+- ``"a"``
+- ``"aaaa"``
+- ``"aba"``
+- ``"racecar"``
+
+Calling ``isPalindrome`` with these inputs should return ``true`` in each case. Notice that these tests are a simple as possible. Keeping test inputs simple, while still covering your desired test cases, will make it easier to fix a but in the event that a unit test failes.
+
+Let's add tests for these inputs to ``spec/palindrome.spec.js``:
+
+.. sourcecode:: js
+   :linenos:
+
+   const isPalindrome = require('../palindrome.js');
+   const assert = require('assert');
+
+   describe("isPalindrome", function(){
+
+      it("should return true for a single letter", function(){
+         assert.strictEqual(isPalindrome("a"), true);
+      });
+
+      it("should return true for a single letter repeated", function(){
+         assert.strictEqual(isPalindrome("aaa"), true);
+      });
+
+      it("should return true for a simple palindrome", function(){
+         assert.strictEqual(isPalindrome("aba"), true);
+      });
+
+      it("should return true for a longer palindrome", function(){
+         assert.strictEqual(isPalindrome("racecar"), true);
+      });
+
+   });
+
+Note the clear test case descriptions (for example, "should return true for a single letter repeated"), which will help us easily identify the expected behavior of our code later.
+
+After adding the positive tests to your file, run them to make sure they all pass.
+
+Negative Test Cases
+^^^^^^^^^^^^^^^^^^^
+
+For ``isPalindrome``, some negative tests have inputs:
+
+- ``"ab"``
+- ``"launchcode"``
+- ``"abA"``
+- ``"so many dynamos"``
+
+Calling ``isPalindrome`` with these inputs should return ``false`` in each case. The last two of these negative tests deserve a bit more discussion.
+
+When writing our ``isPalindrome`` function initially, we made two important decisions:
+
+- Case should be considered, and
+- whitespace should be considered.
+
+The definition of a palindrome differs sometimes on these two matters, so it's important to test them.
+
+Testing with input ``"abA"``` ensures that case is considered, since the lowercase version of this string, ``"aba"``, *is* a palindrome. Testing with ``"so many dynamos"`` ensures that whitespace is considered, since the version of this string with whitespace removed, ``"somanydynamos"``, *is* a palindrome.
+
+.. note:: It's important to isolate your test cases. For example, ``"So Many Dynamos"`` is a poor choice of input for a negative test, since it contains *two* characteristics that are being tested for, case and the presence of whitespace. If a test with this input failed, it would NOT be clear why it failed.
```

Comment:
Grammar: Ended a phrase with a preposition. 
Suggested edit: "...since it tests *two* characteristics - case and whitespace."

Also, word removal: "If a test with this input failed, it would NOT be clear why."